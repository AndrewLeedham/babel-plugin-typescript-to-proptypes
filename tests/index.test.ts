import path from 'path';
import glob from 'glob';
import { transformFileSync } from '@babel/core';
import plugin from '../src';

function transform(filePath: string): string {
  return transformFileSync(filePath, {
    filename: filePath,
    plugins: [plugin],
    generatorOpts: {
      comments: false,
      quotes: 'single',
      jsescOption: { quotes: 'single' },
    },
  }).code;
}

describe('babel-plugin-typescript-to-proptypes', () => {
  glob
    .sync('./fixtures/**/*.ts', { cwd: __dirname, dot: false, strict: true })
    .forEach(filePath => {
      // if (!filePath.endsWith('var/extended-interfaces.ts')) {
      //   return;
      // }

      it(`transforms ${filePath}`, () => {
        expect(transform(path.join(__dirname, filePath))).toMatchSnapshot();
      });
    });
});
