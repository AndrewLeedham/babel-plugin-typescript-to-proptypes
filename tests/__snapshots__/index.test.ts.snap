// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/class/custom-react-import-name.ts 1`] = `
"import PropTypes from 'prop-types';
import R from 'react';
interface Props {
  name: string;
}
export default class ClassCustomReactImportName extends R.Component<Props> {
  static propTypes = {
    name: PropTypes.string.isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/class/custom-react-import-name-pure.ts 1`] = `
"import PropTypes from 'prop-types';
import R from 'react';
interface Props {
  name: string;
}
export default class ClassCustomReactImportNamePure extends R.PureComponent<Props> {
  static propTypes = {
    name: PropTypes.string.isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/class/extended-interfaces.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export interface AProps {
  a: number;
}
export interface BProps {
  b: boolean;
}
export interface Props extends AProps, BProps {
  name: string;
}
export default class ClassExtendedInterfaces extends React.Component<Props> {
  static propTypes = {
    a: PropTypes.number.isRequired,
    b: PropTypes.bool.isRequired,
    name: PropTypes.string.isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/class/extended-type-aliases.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export type AProps = {
  a: number;
};
export type BProps = {
  b: boolean;
};
export type Props = AProps & BProps & {
  name: string;
};
export default class ClassExtendedTypeAliases extends React.Component<Props> {
  static propTypes = {
    a: PropTypes.number.isRequired,
    b: PropTypes.bool.isRequired,
    name: PropTypes.string.isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/class/merge-with-existing-proptypes.ts 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
interface Props {
  name: string;
}
export default class ClassMergeWithExistingPropTypes extends React.Component<Props> {
  static propTypes = {
    name: PropTypes.string.isRequired,
    custom: PropTypes.number.isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/class/merge-with-no-override.ts 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
interface Props {
  name: string;
}
const CustomShape = PropTypes.string;
export default class ClassMergeWithNoOverride extends React.Component<Props> {
  static propTypes = {
    name: CustomShape.isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/class/missing-type.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export default class ClassMissingType extends React.Component<Props> {
  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/class/multiple-generics.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export interface AProps {
  a: number;
}
export interface BProps {
  b: boolean;
}
export default class ClassMultipleGenerics extends React.Component<AProps & BProps> {
  static propTypes = {
    a: PropTypes.number.isRequired,
    b: PropTypes.bool.isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/class/named-react-imports.ts 1`] = `
"import PropTypes from 'prop-types';
import React, { Component } from 'react';
interface Props {
  name: string;
}
export default class ClassNamedReactImports extends Component<Props> {
  static propTypes = {
    name: PropTypes.string.isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/class/named-react-imports-pure.ts 1`] = `
"import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
interface Props {
  name: string;
}
export default class ClassNamedReactImportsPure extends PureComponent<Props> {
  static propTypes = {
    name: PropTypes.string.isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/class/no-generics.ts 1`] = `
"import React from 'react';
export default class ClassNoGenerics extends React.Component {
  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/class/no-type-propeties.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
interface Props {}
export default class ClassNoTypeProperties extends React.Component<Props> {
  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/class/standard.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
interface Props {
  name: string;
}
export default class ClassStandard extends React.Component<Props> {
  static propTypes = {
    name: PropTypes.string.isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/custom-prop-types-import-name.ts 1`] = `
"import React from 'react';
import PT from 'prop-types';
export interface Props {
  name: string;
}
export default class WithCustomPropTypesImportName extends React.Component<Props> {
  static propTypes = {
    name: PT.string.isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/custom-react-import-name.ts 1`] = `
"import PropTypes from 'prop-types';
import R from 'react';
export interface Props {
  name: string;
}
export default class WithCustomReactImportName extends R.Component<Props> {
  static propTypes = {
    name: PropTypes.string.isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/function/custom-react-import-name.ts 1`] = `
"import PropTypes from 'prop-types';
import R from 'react';
interface Props {
  name: string;
}
export default function FuncCustomReactImportName(props: Props) {
  return null;
}
FuncCustomReactImportName.propTypes = {
  name: PropTypes.string.isRequired
};"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/function/export-default.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
interface Props {
  name: string;
}
export default function FuncExportDefault(props: Props) {
  return null;
}
FuncExportDefault.propTypes = {
  name: PropTypes.string.isRequired
};"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/function/export-named.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
interface Props {
  name: string;
}
export function FuncExportNamed(props: Props) {
  return null;
}
FuncExportNamed.propTypes = {
  name: PropTypes.string.isRequired
};"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/function/extended-interfaces.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export interface AProps {
  a: number;
}
export interface BProps {
  b: boolean;
}
export interface Props extends AProps, BProps {
  name: string;
}
export default function FuncExtendedInterfaces(props: Props) {
  return null;
}
FuncExtendedInterfaces.propTypes = {
  a: PropTypes.number.isRequired,
  b: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired
};"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/function/extended-type-aliases.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export type AProps = {
  a: number;
};
export type BProps = {
  b: boolean;
};
export type Props = AProps & BProps & {
  name: string;
};
export default function FuncExtendedTypeAliases(props: Props) {
  return null;
}
FuncExtendedTypeAliases.propTypes = {
  a: PropTypes.number.isRequired,
  b: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired
};"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/function/merge-with-existing-proptypes.ts 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
interface Props {
  name: string;
}
export default function FuncMergeWithExistingPropTypes(props: Props) {
  return null;
}
FuncMergeWithExistingPropTypes.propTypes = {
  name: PropTypes.string.isRequired,
  custom: PropTypes.number.isRequired
};"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/function/merge-with-no-override.ts 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
interface Props {
  name: string;
}
const CustomShape = PropTypes.string;
export default function FuncMergeWithNoOverride(props: Props) {
  return null;
}
FuncMergeWithNoOverride.propTypes = {
  name: CustomShape.isRequired
};"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/function/missing-type.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export default function FuncMissingType(props: Props) {
  return null;
}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/function/multiple-annotations.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export interface AProps {
  a: number;
}
export interface BProps {
  b: boolean;
}
export default function FuncMultipleAnnotations(props: AProps & BProps) {
  return null;
}
FuncMultipleAnnotations.propTypes = {
  a: PropTypes.number.isRequired,
  b: PropTypes.bool.isRequired
};"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/function/no-annotations.ts 1`] = `
"import React from 'react';
export default function FuncNoAnnotations(props) {
  return null;
}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/function/no-export.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
interface Props {
  name: string;
}

function FuncNoExport(props: Props) {
  return null;
}

FuncNoExport.propTypes = {
  name: PropTypes.string.isRequired
};"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/function/no-react-import.ts 1`] = `
"interface Props {}
export default function FuncNoReactImport(props: Props) {
  return null;
}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/function/no-type-propeties.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
interface Props {}
export default function FuncNoTypeProperties(props: Props) {
  return null;
}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/function/standard.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
interface Props {
  name: string;
}
export default function FuncStandard(props: Props) {
  return null;
}
FuncStandard.propTypes = {
  name: PropTypes.string.isRequired
};"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/no-prop-types-import.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export interface Props {
  name: string;
}
export default class NoPropTypesImport extends React.Component<Props> {
  static propTypes = {
    name: PropTypes.string.isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/prop-types-import.ts 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
export interface Props {
  name: string;
}
export default class WithPropTypesImport extends React.Component<Props> {
  static propTypes = {
    name: PropTypes.string.isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/type-array.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export interface Props {
  any: any[];
  null: null[];
  numbers?: number[];
  strings: string[];
  booleans: boolean[];
  objects?: object[];
  union: (string | number)[];
  intersection: (string & number)[];
  nested?: string[][];
}
export default class TypeArray extends React.Component<Props> {
  static propTypes = {
    any: PropTypes.array.isRequired,
    null: PropTypes.array.isRequired,
    numbers: PropTypes.arrayOf(PropTypes.number),
    strings: PropTypes.arrayOf(PropTypes.string).isRequired,
    booleans: PropTypes.arrayOf(PropTypes.bool).isRequired,
    objects: PropTypes.arrayOf(PropTypes.object),
    union: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number])).isRequired,
    intersection: PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.number])).isRequired,
    nested: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string))
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/type-enum.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export interface Props {
  union: number | string | object;
  intersection?: number & string & object;
  combined: number & (string | object);
  strUnion?: 'foo' | 'bar' | 'baz';
  strIntersection: 'foo' & 'bar' & 'baz';
  numUnion: 1 | 2 | 3;
  numIntersection?: 1 & 2 & 3;
}
export default class TypeEnum extends React.Component<Props> {
  static propTypes = {
    union: PropTypes.oneOfType([PropTypes.number, PropTypes.string, PropTypes.object]).isRequired,
    intersection: PropTypes.oneOfType([PropTypes.number, PropTypes.string, PropTypes.object]),
    combined: PropTypes.oneOfType([PropTypes.number, PropTypes.oneOfType([PropTypes.string, PropTypes.object])]).isRequired,
    strUnion: PropTypes.oneOf(['foo', 'bar', 'baz']),
    strIntersection: PropTypes.oneOf(['foo', 'bar', 'baz']).isRequired,
    numUnion: PropTypes.oneOf([1, 2, 3]).isRequired,
    numIntersection: PropTypes.oneOf([1, 2, 3])
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/type-function.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export interface Props {
  void: () => void;
  objReturn: () => {};
  typeReturn?: () => string;
  args: (a: number, b: boolean) => null;
  parens?: (() => void);
  handler: React.ChangeEventHandler;
}
export default class TypeFunction extends React.Component<Props> {
  static propTypes = {
    void: PropTypes.func.isRequired,
    objReturn: PropTypes.func.isRequired,
    typeReturn: PropTypes.func,
    args: PropTypes.func.isRequired,
    parens: PropTypes.func,
    handler: PropTypes.func.isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/type-object.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export interface Props {
  object: object;
  empty: {};
  index?: {
    [key: string]: number;
  };
  shape: {
    id: number;
    name?: string;
    status: 'active' | 'pending';
  };
}
export default class TypeObject extends React.Component<Props> {
  static propTypes = {
    object: PropTypes.object.isRequired,
    empty: PropTypes.object.isRequired,
    index: PropTypes.objectOf(PropTypes.number),
    shape: PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string,
      status: PropTypes.oneOf(['active', 'pending']).isRequired
    }).isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/type-primitives.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export interface Props {
  number: number;
  string: string;
  boolean: boolean;
  symbol: symbol;
  numberOpt?: number;
  stringOpt?: string;
  booleanOpt?: boolean;
  symbolOpt?: symbol;
}
export default class TypePrimitives extends React.Component<Props> {
  static propTypes = {
    number: PropTypes.number.isRequired,
    string: PropTypes.string.isRequired,
    boolean: PropTypes.bool.isRequired,
    symbol: PropTypes.symbol.isRequired,
    numberOpt: PropTypes.number,
    stringOpt: PropTypes.string,
    booleanOpt: PropTypes.bool,
    symbolOpt: PropTypes.symbol
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/type-react.ts 1`] = `
"import PropTypes from 'prop-types';
import React, { ReactText, ReactNode, ReactType, ComponentType, ComponentClass, StatelessComponent, MouseEvent, ReactElement, SFCElement } from 'react';
export interface Props {
  text: React.ReactText;
  textAlias?: ReactText;
  node: React.ReactNode;
  nodeAlias?: ReactNode;
  type: React.ReactType;
  typeAlias?: ReactType;
  comp: React.ComponentType;
  compGeneric?: React.ComponentType<any>;
  compAlias: ComponentType;
  cls: React.ComponentClass;
  clsGeneric?: React.ComponentClass<any>;
  clsAlias: ComponentClass;
  sfc: React.StatelessComponent;
  sfcGeneric?: React.StatelessComponent<any>;
  sfcAlias: StatelessComponent;
  el: React.ReactElement<any>;
  elAlias?: ReactElement<any>;
  sfcEl: React.SFCElement<any>;
  sfcElAlias?: SFCElement<any>;
  jsx: JSX.Element;
  event: React.MouseEvent;
  eventAlias?: MouseEvent;
  ref: React.Ref<any>;
}
export default class TypeReact extends React.Component<Props> {
  static propTypes = {
    text: PropTypes.node.isRequired,
    textAlias: PropTypes.node,
    node: PropTypes.node.isRequired,
    nodeAlias: PropTypes.node,
    type: PropTypes.node.isRequired,
    typeAlias: PropTypes.node,
    comp: PropTypes.node.isRequired,
    compGeneric: PropTypes.node,
    compAlias: PropTypes.node.isRequired,
    cls: PropTypes.node.isRequired,
    clsGeneric: PropTypes.node,
    clsAlias: PropTypes.node.isRequired,
    sfc: PropTypes.node.isRequired,
    sfcGeneric: PropTypes.node,
    sfcAlias: PropTypes.node.isRequired,
    el: PropTypes.element.isRequired,
    elAlias: PropTypes.element,
    sfcEl: PropTypes.element.isRequired,
    sfcElAlias: PropTypes.element,
    jsx: PropTypes.element.isRequired,
    event: PropTypes.object.isRequired,
    eventAlias: PropTypes.object,
    ref: PropTypes.oneOfType([PropTypes.string, PropTypes.func, PropTypes.object]).isRequired
  };

  render() {
    return null;
  }

}"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/var/custom-react-import-name.ts 1`] = `
"import R from 'react';
interface Props {
  name: string;
}

const VarCustomReactImportName: React.SFC<Props> = () => null;

export default VarCustomReactImportName;"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/var/export-default.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
interface Props {
  name: string;
}

const VarExportDefault: React.SFC<Props> = () => null;

VarExportDefault.propTypes = {
  name: PropTypes.string.isRequired
};
export default VarExportDefault;"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/var/export-named.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
interface Props {
  name: string;
}
export const VarExportNamed: React.SFC<Props> = () => null;
VarExportNamed.propTypes = {
  name: PropTypes.string.isRequired
};"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/var/extended-interfaces.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export interface AProps {
  a: number;
}
export interface BProps {
  b: boolean;
}
export interface Props extends AProps, BProps {
  name: string;
}

const VarExtendedInterfaces: React.SFC<Props> = () => null;

VarExtendedInterfaces.propTypes = {
  a: PropTypes.number.isRequired,
  b: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired
};
export default VarExtendedInterfaces;"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/var/extended-type-aliases.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export type AProps = {
  a: number;
};
export type BProps = {
  b: boolean;
};
export type Props = AProps & BProps & {
  name: string;
};

const VarExtendedTypeAliases: React.SFC<Props> = () => null;

VarExtendedTypeAliases.propTypes = {
  a: PropTypes.number.isRequired,
  b: PropTypes.bool.isRequired,
  name: PropTypes.string.isRequired
};
export default VarExtendedTypeAliases;"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/var/merge-with-existing-proptypes.ts 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
interface Props {
  name: string;
}

const VarMergeWithExistingPropTypes: React.SFC<Props> = () => null;

VarMergeWithExistingPropTypes.propTypes = {
  name: PropTypes.string.isRequired,
  custom: PropTypes.number.isRequired
};
export default VarMergeWithExistingPropTypes;"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/var/merge-with-no-override.ts 1`] = `
"import React from 'react';
import PropTypes from 'prop-types';
interface Props {
  name: string;
}
const CustomShape = PropTypes.string;

const VarMergeWithNoOverride: React.SFC<Props> = () => null;

VarMergeWithNoOverride.propTypes = {
  name: CustomShape.isRequired
};
export default VarMergeWithNoOverride;"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/var/missing-type.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';

const VarMissingType: React.SFC<Props> = () => null;

export default VarMissingType;"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/var/multiple-annotations.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
export interface AProps {
  a: number;
}
export interface BProps {
  b: boolean;
}

const VarMultipleAnnotations: React.SFC<AProps & BProps> = () => null;

VarMultipleAnnotations.propTypes = {
  a: PropTypes.number.isRequired,
  b: PropTypes.bool.isRequired
};
export default VarMultipleAnnotations;"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/var/no-annotations.ts 1`] = `
"import React from 'react';

const VarNoAnnotations: React.SFC = () => null;

export default VarNoAnnotations;"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/var/no-export.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
interface Props {
  name: string;
}

const VarNoExport: React.SFC<Props> = () => null;

VarNoExport.propTypes = {
  name: PropTypes.string.isRequired
};"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/var/no-react-import.ts 1`] = `
"interface Props {}

const VarNoReactImport: React.SFC<Props> = () => null;

export default VarNoReactImport;"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/var/no-type-propeties.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
interface Props {}

const VarNoTypeProperties: React.SFC<Props> = () => null;

export default VarNoTypeProperties;"
`;

exports[`babel-plugin-typescript-to-proptypes transforms ./fixtures/var/standard.ts 1`] = `
"import PropTypes from 'prop-types';
import React from 'react';
interface Props {
  name: string;
}

const VarStandard: React.SFC<Props> = () => null;

VarStandard.propTypes = {
  name: PropTypes.string.isRequired
};
export default VarStandard;"
`;
